### PR TITLE
Do not run value transformation if a option is being requested by its name and there is no dependency on other options

### LINKED ISSUE #24757: DefaultHostnameProvider reports being proxied when it is not
On the DefaultHostnameProvider log, that appears to be an incomplete fix from https://github.com/keycloak/keycloak/issues/10627 / https://github.com/keycloak/keycloak/pull/10707 - if kc.proxy is not not set, then false is being returned as the value, but if you explicitly set none, then none is being returned and noProxy is set to false.

cc @vmuzikar @pedroigor PropertyMapper.transformValue will return a ConfigValue representing the default where the raw value here is "none", but the mapped value is false.  This makes it inconsistent when explicitly using --proxy=none as the mapped value will be none.  It seems like there may need to be a change with the default handling, or the logic in the DefaultHostnameProvider will need to check the raw value instead.

_Originally posted by @shawkins in https://github.com/keycloak/keycloak/issues/24294#issuecomment-1808653896_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #24757

### CODE REVIEW

File: quarkus/config-api/src/main/java/org/keycloak/config/ProxyOptions.java [hunk 1]
```diff
@@ -3,10 +3,24 @@
...
+
+        Mode(boolean proxyEnabled) {
+            this.proxyEnabled = proxyEnabled;
+        }
+
+        Mode() {
+            this(true);
+        }
+
+        public boolean isProxyEnabled() {
```
Comment (vmuzikar) [inline quarkus/config-api/src/main/java/org/keycloak/config/ProxyOptions.java:21]: I'd rename this to `isProxyHeadersEnabled` as `passthrough` is still a proxy mode and e.g. should use 443 port.

**Review by vmuzikar — APPROVED** [review]
@pedroigor Thanks, LGTM.

@shawkins Do you want to give it a final look too?

**Comment by pedroigor** [comment]
@vmuzikar @shawkins Deep sorries ... I went to quick and don't see the comment from Vasek asking for Steven review.

I can either revert or do follow-ups depending on @shawkins feedback. Sorry.

**Comment by shawkins** [comment]
@pedroigor no additional concerns from me.  Thank you for untying this knot.

**Pull request merged**